docs/index.md

```diff
@@ -1,49 +1,40 @@
+UCC 5 <!--- why do we say 5, everywhere else we say UCC---> uses Splunk UI. See [Splunk UI](https://splunkui.splunk.com/). It is a new UI framework based on React. The UCC UI repository is stored in the `ui` folder. <!-- (but where is this folder exactly, when I install Splunk UI?) -->
 
-You can use [Splunk Extension for VSCode](https://marketplace.visualstudio.com/items?itemName=Splunk.splunk)
-as well.
+> **Note:** The UCC framework supports add-ons that use only Python 3.
 
-## What is UCC?
+The UCC framework is available as a GitHub action. See <https://github.com/splunk/addonfactory-ucc-generator-action>.
 
-UCC stands for Universal Configuration Console. The purpose of having a
-framework for add-on generation is to simplify the process of add-on
-creation for developers. UCC 5 uses [SplunkUI](https://splunkui.splunk.com/),
-which is a new UI framework based on React. The UCC UI repository can be found in the `ui` folder.
+To work with UCC framework, you can also use Splunk Extension. It helps you to create, test, and debug the add-ons in a simple way. For more information, see [Visual Studio Code Extension for Splunk](https://marketplace.visualstudio.com/items?itemName=Splunk.splunk).
 
-UCC-based add-ons are being powered by Splunk libraries:
-[`solnlib`](https://github.com/splunk/addonfactory-solutions-library-python) and
-[`splunktaucclib`](https://github.com/splunk/addonfactory-ucc-library). More
-information [here](ucc_related_libraries.md).
+## Libraries
 
-## Features
+UCC-based add-ons are powered by the following Splunk libraries:
+* `solnlib`, see [https://github.com/splunk/addonfactory-solutions-library-python](https://github.com/splunk/addonfactory-solutions-library-python)
+* `splunktaucclib`, see [https://github.com/splunk/addonfactory-ucc-library](https://github.com/splunk/addonfactory-ucc-library). 
 
-The `splunk-add-on-ucc-framework`:
+For more information, see [UCC-related libraries](ucc_related_libraries.md).
 
-* generates UI (`appserver` folder).
-* generates Python REST handlers to support UI CRUD operations (`bin` folder).
-* generates [inputs](./inputs/index.md) and their [helper modules](./inputs/helper.md)
-* generates OpenAPI description documents (`appserver/static/openapi.json` file) (for more information, see [here](openapi.md)).
-* generates `.conf` files (more information, see [here](dot_conf_files.md)).
-* installs Python requirements (`lib` folder).
-* generate metadata files (`metadata` folder).
-* generates the monitoring dashboard (for more information, see [here](dashboard.md)).
-* it possibly extends the UI with custom codes (for more information, see [here](custom_ui_extensions/custom_hook.md)).
-* it possibly extends the build process via a `additional_packaging.py` file (more information, [here](additional_packaging.md)).
-* generates the necessary files defined for the Alert Action, if defined in globalConfig (for more informaiton, see [here](alert_actions/index.md)).
+## What UCC generates 
```

Comment:
I added the caveats as a note in this section. @artemrys do we leave this info here?

Reply:
Yes, let's leave it for now, that's not a deal breaker for me.